services/m3aggregator/main/main.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/m3db/m3aggregator/aggregator"
+	httpserver "github.com/m3db/m3aggregator/server/http"
+	msgpackserver "github.com/m3db/m3aggregator/server/msgpack"
+	"github.com/m3db/m3aggregator/services/m3aggregator/processor"
+	"github.com/m3db/m3aggregator/services/m3aggregator/serve"
+	"github.com/m3db/m3metrics/metric/aggregated"
+	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3x/log"
+)
+
+const (
+	gracefulShutdownTimeout = 10 * time.Second
+)
+
+var (
+	msgpackAddrArg = flag.String("msgpackAddr", "0.0.0.0:6000", "msgpack server address")
```

Comment:
These are used to stand up an example server quickly. In production we'll have a different main file internally that reads from config files.